src/Microsoft.Extensions.Configuration.UserSecrets/ConfigurationExtensions.cs

```diff
@@ -28,11 +28,18 @@ public static IConfigurationBuilder AddUserSecrets<T>(this IConfigurationBuilder
             => configuration.AddUserSecrets(typeof(T).GetTypeInfo().Assembly);
 
         /// <summary>
+        /// <para>
+        /// This method is obsolete and will be removed in a future version.
+        /// The recommended alternative is <see cref="AddUserSecrets(IConfigurationBuilder, string)"/>.
+        /// </para>
+        /// <para>
         /// Adds the user secrets configuration source. Searches the assembly from <see cref="Assembly.GetEntryAssembly"/>
         /// for an instance of <see cref="UserSecretsIdAttribute"/>.
+        /// </para>
         /// </summary>
         /// <param name="configuration"></param>
         /// <returns></returns>
+        [Obsolete("This method is obsolete and will be removed in a future version. The recommended alternative is .AddUserSecrets(string userSecretsId).")]
```

Comment:
> @divega the part of the conversation that you missed is that we were talking about #575 . We figured it is better for users to upgrade to the .AddUserSecrets(string) overload since there is a chance we will remove the .AddUserSecrets<T>() overload in 2.0. You make a good point though. Pain now or possible pain later?

FWIW, my take is that the approach described on #575 is valid... It was probably a good idea to switch to that some time ago, and it will probably still be a good idea to try to make the switch in the future. But doing it right now seems to charge the cost of the switch to the bill of expenses of the project.json to .csproj migration, which is a bit too onerous by itself.